src/cpu.rs

```diff
@@ -162,6 +162,27 @@ pub(crate) mod arm {
         }
     }
 
+    #[cfg(all(target_os = "windows", target_arch = "aarch64"))]
+    pub fn setup() {
+        // We do not need to check for the presence of NEON, as Armv8-A always has it
+        let mut features = NEON.mask;
+
+        let result = unsafe {
+            winapi::um::processthreadsapi::IsProcessorFeaturePresent(
+                winapi::um::winnt::PF_ARM_V8_CRYPTO_INSTRUCTIONS_AVAILABLE
+            )
+        };
+
+        if result != 0 {
+            // These are all covered by one call in Windows
+            features |= AES.mask;
+            features |= PMULL.mask;
+            features |= SHA256.mask;
```

Comment:
Do I understand correctly that \*ring\* doesn't support SHA1? (see [BoringSSL patch](https://github.com/google/boringssl/commit/afd5dba756b6266fa99c11af6496b39d826769cd#diff-86205ea0f68c5c51fb60ca69d8da418f3f6bb855816ae047227ed69853c75eaaR36))

Reply:
Correct. *ring* currently doesn't use accelerated SHA-1. We will probably change that soon, as now there are commercial users that do need the SHA-1 to be fast. That's future work though.

Reply:
Would you prefer to leave a TODO here so we don't forget to enable it later on?

Reply:
No, because the other code paths also need to be modified if/when we do that.